### ISSUE TITLE
build --no-directory-urls feature not working

### ISSUE THREAD

**Issue by Drgzidjiboy4ik** [mkdocs/mkdocs#2363]
Hello. 
I created new project using 
```
mkdocs new .
mkdocs build --no-directory-urls
```

Still when I try to visit home page I am brought to . directory instead of .\index.html

Sorry for some russian text :D
![image](https://user-images.githubusercontent.com/3143647/114592047-3055ff80-9c93-11eb-9477-e1beabc8fa66.png)

I also tried to modife .yml file like this 
`use_directory_urls: false`

Still not working in "offline" mode

```
pip show mkdocs
Name: mkdocs
Version: 1.1.2
Summary: Project documentation with Markdown.
Home-page: https://www.mkdocs.org
Author: Tom Christie
Author-email: tom@tomchristie.com
License: BSD
Location: /home/demaunt/.local/lib/python3.8/site-packages
Requires: Markdown, lunr, Jinja2, PyYAML, click, livereload, tornado
Required-by: mkdocs-redirects, mkdocs-minify-plugin, mkdocs-material
```

**Comment by waylan** [comment]
I can verify this is a bug. 

![Unknown](https://user-images.githubusercontent.com/78846/114794973-91490a80-9d5b-11eb-8936-e4ce1c29d283.png)

In the above screen capture, the link circled in red is wrong (`href="."`), whereas the link circled in yellow is correct (`href="index.html"`). Although, I should note that both work fine when behind a server. The first is only broken when browsing local files (`file:///`).

It appears that the issue is related to the `homepage`. The link is created from `nav.homepage`. Yet, I note that when `use_directory_urls` is `False`, the `homepage` class is also missing from the body tag of the homepage. It seems that the homepage is not properly identified.

**Comment by Drgzidjiboy4ik** [comment]
Is this issue fixed? Or shoild I wait until 1.2 is released?

**Comment by waylan** [comment]
Yes, it is fixed in that the fix is in the master branch and it will be available in the next release (1.2).
